src/main/io/gps.c

```diff
@@ -957,33 +1485,47 @@ void gpsUpdate(timeUs_t currentTimeUs)
     DEBUG_SET(DEBUG_GPS_DOP, 3, gpsSol.dop.vdop);
 
     executeTimeUs = micros() - currentTimeUs;
-    if (executeTimeUs > gpsStateDurationUs[gpsCurrentState]) {
-        gpsStateDurationUs[gpsCurrentState] = executeTimeUs;
+    if (executeTimeUs > (gpsStateDurationFractionUs[gpsCurrentState] >> GPS_TASK_DECAY_SHIFT)) {
+        gpsStateDurationFractionUs[gpsCurrentState] += (2 << GPS_TASK_DECAY_SHIFT);
+    } else {
+        // Slowly decay the max time
+        gpsStateDurationFractionUs[gpsCurrentState]--;
```

Comment:
This will decay at 1/512 us/call, roughly 5s to decay by single us at 100Hz. Is that intentional?

Reply:
Yes, a slow decay is intended.  However it isn't as slow as expected, in practice, because every time we get incoming data, we run at 500hz.  Typically a Nav packet results in 4-5 fast tasks, and that happens every 10 slow tasks, so the decay time is about 3-4 seconds to decay a single us at 115200Hz.  At slower baud rates, the proportion of fast to slow tasks increases.  At 9600 baud and 5hz, or 19200 and 10hz, we spend more than 50% of our time at 500hz, so the decay time then is then of the order of 1-2s.  In any case, as slow decay is intended, since as the scheduler limit value drops, we will eventually hit it, and that will be recorded as a 'late task'.  So we don't want the scheduler limit to drop really fast, nor really slow.  I tested and measured this extensively with @SteveCEvans and it works very well, in practice.